{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "LogonFailure",
     "evalue": "Received unexpected status from the server: The attempted logon is invalid. This is either due to a bad username or authentication information. (3221225581) STATUS_LOGON_FAILURE: 0xc000006d",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mLogonFailure\u001b[0m                              Traceback (most recent call last)",
      "\u001b[1;32mc:\\Users\\aksel\\Desktop\\Artefact_Detection\\cluster_test.ipynb Cell 1'\u001b[0m in \u001b[0;36m<cell line: 8>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/aksel/Desktop/Artefact_Detection/cluster_test.ipynb#ch0000000?line=2'>3</a>\u001b[0m smbclient\u001b[39m.\u001b[39mClientConfig(username\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mau568033@uni.au.dk\u001b[39m\u001b[39m'\u001b[39m, password\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39miKkel1gefremdec1deret\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/aksel/Desktop/Artefact_Detection/cluster_test.ipynb#ch0000000?line=4'>5</a>\u001b[0m \u001b[39m# Optional - register the credentials with a server (overrides ClientConfig for that server)\u001b[39;00m\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/aksel/Desktop/Artefact_Detection/cluster_test.ipynb#ch0000000?line=5'>6</a>\u001b[0m \u001b[39m#smbclient.register_session(\"server\", username=\"user\", password=\"pass\")\u001b[39;00m\n\u001b[1;32m----> <a href='vscode-notebook-cell:/c%3A/Users/aksel/Desktop/Artefact_Detection/cluster_test.ipynb#ch0000000?line=7'>8</a>\u001b[0m smbclient\u001b[39m.\u001b[39;49mmkdir(\u001b[39mr\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39m//uni.au.dk/dfs\u001b[39;49m\u001b[39m\"\u001b[39;49m,username\u001b[39m=\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39mau568033@uni.au.dk\u001b[39;49m\u001b[39m'\u001b[39;49m, password\u001b[39m=\u001b[39;49m\u001b[39m'\u001b[39;49m\u001b[39miKkel1gefremdec1deret\u001b[39;49m\u001b[39m'\u001b[39;49m)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbclient\\_os.py:267\u001b[0m, in \u001b[0;36mmkdir\u001b[1;34m(path, **kwargs)\u001b[0m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=259'>260</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mmkdir\u001b[39m(path, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs):\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=260'>261</a>\u001b[0m     \u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=261'>262</a>\u001b[0m \u001b[39m    Create a directory named path. If the directory already exists, OSError(errno.EEXIST) is raised.\u001b[39;00m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=262'>263</a>\u001b[0m \n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=263'>264</a>\u001b[0m \u001b[39m    :param path: The path to the directory to create.\u001b[39;00m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=264'>265</a>\u001b[0m \u001b[39m    :param kwargs: Common SMB Session arguments for smbclient.\u001b[39;00m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=265'>266</a>\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=266'>267</a>\u001b[0m     raw \u001b[39m=\u001b[39m SMBDirectoryIO(path, mode\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mx\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=267'>268</a>\u001b[0m     \u001b[39mwith\u001b[39;00m SMBFileTransaction(raw):\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_os.py?line=268'>269</a>\u001b[0m         \u001b[39mpass\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbclient\\_io.py:377\u001b[0m, in \u001b[0;36mSMBRawIO.__init__\u001b[1;34m(self, path, mode, share_access, desired_access, file_attributes, create_options, **kwargs)\u001b[0m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_io.py?line=374'>375</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m__init__\u001b[39m(\u001b[39mself\u001b[39m, path, mode\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mr\u001b[39m\u001b[39m'\u001b[39m, share_access\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, desired_access\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, file_attributes\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m,\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_io.py?line=375'>376</a>\u001b[0m              create_options\u001b[39m=\u001b[39m\u001b[39m0\u001b[39m, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs):\n\u001b[1;32m--> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_io.py?line=376'>377</a>\u001b[0m     tree, fd_path \u001b[39m=\u001b[39m get_smb_tree(path, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_io.py?line=377'>378</a>\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mshare_access \u001b[39m=\u001b[39m share_access\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_io.py?line=378'>379</a>\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mfd \u001b[39m=\u001b[39m Open(tree, fd_path)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbclient\\_pool.py:310\u001b[0m, in \u001b[0;36mget_smb_tree\u001b[1;34m(path, username, password, port, encrypt, connection_timeout, connection_cache)\u001b[0m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=306'>307</a>\u001b[0m         path_split \u001b[39m=\u001b[39m [p \u001b[39mfor\u001b[39;00m p \u001b[39min\u001b[39;00m path\u001b[39m.\u001b[39msplit(\u001b[39m\"\u001b[39m\u001b[39m\\\\\u001b[39;00m\u001b[39m\"\u001b[39m) \u001b[39mif\u001b[39;00m p]\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=308'>309</a>\u001b[0m server \u001b[39m=\u001b[39m path_split[\u001b[39m0\u001b[39m]\n\u001b[1;32m--> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=309'>310</a>\u001b[0m session \u001b[39m=\u001b[39m register_session(server, username\u001b[39m=\u001b[39;49musername, password\u001b[39m=\u001b[39;49mpassword, port\u001b[39m=\u001b[39;49mport, encrypt\u001b[39m=\u001b[39;49mencrypt,\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=310'>311</a>\u001b[0m                            connection_timeout\u001b[39m=\u001b[39;49mconnection_timeout, connection_cache\u001b[39m=\u001b[39;49mconnection_cache,\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=311'>312</a>\u001b[0m                            auth_protocol\u001b[39m=\u001b[39;49mauth_protocol)\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=313'>314</a>\u001b[0m share_path \u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39m\\\\\u001b[39;00m\u001b[39m\\\\\u001b[39;00m\u001b[39m%s\u001b[39;00m\u001b[39m\\\\\u001b[39;00m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m (server, path_split[\u001b[39m1\u001b[39m])\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=314'>315</a>\u001b[0m tree \u001b[39m=\u001b[39m \u001b[39mnext\u001b[39m((t \u001b[39mfor\u001b[39;00m t \u001b[39min\u001b[39;00m session\u001b[39m.\u001b[39mtree_connect_table\u001b[39m.\u001b[39mvalues() \u001b[39mif\u001b[39;00m t\u001b[39m.\u001b[39mshare_name \u001b[39m==\u001b[39m share_path), \u001b[39mNone\u001b[39;00m)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbclient\\_pool.py:383\u001b[0m, in \u001b[0;36mregister_session\u001b[1;34m(server, username, password, port, encrypt, connection_timeout, connection_cache, auth_protocol, require_signing)\u001b[0m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=379'>380</a>\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m session:\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=380'>381</a>\u001b[0m     session \u001b[39m=\u001b[39m Session(connection, username\u001b[39m=\u001b[39musername, password\u001b[39m=\u001b[39mpassword, require_encryption\u001b[39m=\u001b[39m(encrypt \u001b[39mis\u001b[39;00m \u001b[39mTrue\u001b[39;00m),\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=381'>382</a>\u001b[0m                       auth_protocol\u001b[39m=\u001b[39mauth_protocol)\n\u001b[1;32m--> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=382'>383</a>\u001b[0m     session\u001b[39m.\u001b[39;49mconnect()\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=383'>384</a>\u001b[0m \u001b[39melif\u001b[39;00m encrypt \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=384'>385</a>\u001b[0m     \u001b[39m# We cannot go from encryption to no encryption on an existing session but we can do the opposite.\u001b[39;00m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbclient/_pool.py?line=385'>386</a>\u001b[0m     \u001b[39mif\u001b[39;00m session\u001b[39m.\u001b[39mencrypt_data \u001b[39mand\u001b[39;00m \u001b[39mnot\u001b[39;00m encrypt:\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbprotocol\\session.py:295\u001b[0m, in \u001b[0;36mSession.connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/session.py?line=292'>293</a>\u001b[0m log\u001b[39m.\u001b[39minfo(\u001b[39m\"\u001b[39m\u001b[39mReceiving SMB2_SESSION_SETUP response message\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/session.py?line=293'>294</a>\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m--> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/session.py?line=294'>295</a>\u001b[0m     response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mconnection\u001b[39m.\u001b[39;49mreceive(request)\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/session.py?line=295'>296</a>\u001b[0m \u001b[39mexcept\u001b[39;00m MoreProcessingRequired \u001b[39mas\u001b[39;00m exc:\n\u001b[0;32m    <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/session.py?line=296'>297</a>\u001b[0m     mid \u001b[39m=\u001b[39m request\u001b[39m.\u001b[39mmessage[\u001b[39m'\u001b[39m\u001b[39mmessage_id\u001b[39m\u001b[39m'\u001b[39m]\u001b[39m.\u001b[39mget_value()\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\smbprotocol\\connection.py:1006\u001b[0m, in \u001b[0;36mConnection.receive\u001b[1;34m(self, request, wait, timeout, resolve_symlinks)\u001b[0m\n\u001b[0;32m   <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/connection.py?line=1002'>1003</a>\u001b[0m             \u001b[39mself\u001b[39m\u001b[39m.\u001b[39moutstanding_requests\u001b[39m.\u001b[39mpop(message_id, \u001b[39mNone\u001b[39;00m)\n\u001b[0;32m   <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/connection.py?line=1004'>1005</a>\u001b[0m             \u001b[39mif\u001b[39;00m status \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m [NtStatus\u001b[39m.\u001b[39mSTATUS_SUCCESS, NtStatus\u001b[39m.\u001b[39mSTATUS_PENDING]:\n\u001b[1;32m-> <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/connection.py?line=1005'>1006</a>\u001b[0m                 \u001b[39mraise\u001b[39;00m SMBResponseException(response)\n\u001b[0;32m   <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/connection.py?line=1007'>1008</a>\u001b[0m             \u001b[39mbreak\u001b[39;00m\n\u001b[0;32m   <a href='file:///~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/site-packages/smbprotocol/connection.py?line=1009'>1010</a>\u001b[0m \u001b[39mreturn\u001b[39;00m response\n",
      "\u001b[1;31mLogonFailure\u001b[0m: Received unexpected status from the server: The attempted logon is invalid. This is either due to a bad username or authentication information. (3221225581) STATUS_LOGON_FAILURE: 0xc000006d"
     ]
    }
   ],
   "source": [
    "import smbclient\n",
    "\n",
    "smbclient.ClientConfig(username='au568033@uni.au.dk', password='iKkel1gefremdec1deret8')\n",
    "\n",
    "# Optional - register the credentials with a server (overrides ClientConfig for that server)\n",
    "#smbclient.register_session(\"server\", username=\"user\", password=\"pass\")\n",
    "\n",
    "smbclient.mkdir(r\"//uni.au.dk/dfs\",username='au568033@uni.au.dk', password='iKkel1gefremdec1deret8')\n",
    "\n",
    "#with smbclient.open_file(r\"\\\\server\\share\\directory\\file.txt\", mode=\"w\") as fd:\n",
    "#    fd.write(u\"file contents\")\n",
    "\n",
    "\n",
    "\n",
    "#smb://uni.au.dk/dfs"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "635667bee88eb83c325aa716129aeacb0651bf4a2192bd0dd3c5b7b0ba12955b"
  },
  "kernelspec": {
   "display_name": "Python 3.9.10 64-bit (windows store)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.10"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
